packages/@react-spectrum/combobox/stories/ComboBox.stories.tsx

```diff
@@ -0,0 +1,416 @@
+    () => (
+      <ControlledKeyComboBox defaultInputValue="Blah" selectedKey="2" />
+    )
+  )
+  .add(
+    'isQuiet',
+    () => render({isQuiet: true})
+  )
+  .add(
+    'isDisabled',
+    () => render({isDisabled: true})
+  )
+  .add(
+    'isReadOnly',
+    () => render({isReadOnly: true})
+  )
+  .add(
+    'labelPosition: top, labelAlign: end',
+    () => render({labelAlign: 'end'})
+  )
+  .add(
+    'labelPosition: side',
+    () => render({labelPosition: 'side'})
+  )
+  .add(
+    'isRequired',
+    () => render({isRequired: true})
+  )
+  .add(
+    'isRequired, necessityIndicator: label',
+    () => render({isRequired: true, necessityIndicator: 'label'})
+  )
+  .add(
+    'validationState: invalid',
+    () => render({validationState: 'invalid', defaultSelectedKey: 'two'})
+  )
+  .add(
+    'validationState: valid',
+    () => render({validationState: 'valid', defaultSelectedKey: 'two'})
+  )
+  .add(
+    'autoFocus: true',
+    () => render({autoFocus: true})
+  )
+  .add(
+    'direction: top',
+    () => render({direction: 'top'})
+  )
+  .add(
+    'allowsCustomValue: true',
+    () => (
+      <CustomValueComboBox allowsCustomValue selectedKey="2" />
+    )
+  )
+  .add(
+    'customWidth',
+    () => (
+      <Flex direction="column">
+        <ComboBox label="Combobox" {...actions} width="200px">
+          <Item key="one">Item One</Item>
+          <Item key="two" textValue="Item Two">
+            <Copy size="S" />
+            <Text>Item Two</Text>
+          </Item>
+          <Item key="three">Item Three</Item>
+        </ComboBox>
+        <ComboBox label="Combobox" {...actions} width="800px">
+          <Item key="one">Item One</Item>
+          <Item key="two" textValue="Item Two">
+            <Copy size="S" />
+            <Text>Item Two</Text>
+          </Item>
+          <Item key="three">Item Three</Item>
+        </ComboBox>
+      </Flex>
+    )
+  )
+  .add(
+    'onFilter, (value included in list)',
+    () => (
+      <CustomFilterComboBox selectedKey="2" />
+    )
+  );
+
+let customFilterItems = [
+  {name: 'The first item', id: '1'},
+  {name: 'The second item', id: '2'},
+  {name: 'The third item', id: '3'}
+];
+
+let CustomFilterComboBox = (props) => {
+  let [list, setList] = React.useState(customFilterItems);
+  let [selectedKey, setSelectedKey] = React.useState(props.selectedKey);
+
+  let onSelectionChange = (key) => {
+    setSelectedKey(key);
+  };
+
+  let onFilter = (value) => {
+    setList(customFilterItems.filter(item => item.name.includes(value)));
+  };
+
+  return (
+    <ComboBox items={list} selectedKey={selectedKey} label="Combobox" onSelectionChange={onSelectionChange} onFilter={onFilter} onOpenChange={action('onOpenChange')} onInputChange={action('onInputChange')} onBlur={action('onBlur')} onFocus={action('onFocus')} onCustomValue={action('onCustomValue')}>
+      {(item: any) => <Item>{item.name}</Item>}
+    </ComboBox>
+  );
+};
+
+let AllControlledComboBox = (props) => {
```

Comment:
Bit concerned about how users would use a "controlled key and controlled input value" ComboBox and whether or not they may run into the error thrown when the `props.selectedKey` and `props.inputValue` don't match. In a set up like below, how would they have a button that updates the key to "6" and the inputValue to "Ross", would they need to batch the updates?

I had something like this here originally
```
      <ButtonGroup marginEnd="30px">
        <Button variant="secondary" onPress={() => {
          setInputValue('Snake')
          setSelectedKey('3')
        }}>
          <Text>Ross</Text>
        </Button>
        <Button variant="secondary" onPress={() => {
          setInputValue('Ross')
          setSelectedKey('6')
        }}>
          <Text>Ross</Text>
        </Button>
      </ButtonGroup>
```
 but ran into the above issue